middle_end/flambda/types/env/typing_env.rec.ml

```diff
@@ -1266,11 +1265,22 @@ let type_simple_in_term_exn t ?min_name_mode simple =
         Name_mode.normal
     in
     let [@inline always] name name ~coercion:_ =
```

Comment:
Same here. In my mind, when a function takes an argument it means that it will get used. Here's there's a couple of [name] function that take a [coercion] but does nothing with it (apart from checking if it's [Id])

Reply:
Yeah, same as above. Needed to satisfy `pattern_match` API.